lib/async/io/shared_endpoint.rb

```diff
@@ -26,8 +26,8 @@ class SharedEndpoint < Endpoint
 			def self.bound(endpoint, backlog = Socket::SOMAXCONN)
 				wrappers = []
 				
-				endpoint.each do |endpoint|
```

Comment:
I dislike abbreviations, especially in situations like this because you are abbreviating it just to avoid name collision. Abbreviations also make code less accessible (i.e. visually impaired, non-native speaker, etc).

I prefer finding some way to resolve this ambiguity which makes the intention of the code more clear.

This is a tricky one because there is not a clean semantic difference between an endpoint which represents multiple "points of contact" and a single point of contact. You call `endpoint.each` and you basically get more endpoints, although there is a guarantee that if you have an plural endpoint, you'll get singular endpoints as a result.

Based on that, maybe this is better:

```
endpoint.each do |singular_endpoint|
end
```